File: engine/contracts/session_ref.py
"""此文件由 scripts/generate-contracts.py 生成，请勿手改。"""
from __future__ import annotations

OPAQUE_SESSION_REF_PREFIX = 'fsr_'
OPAQUE_SESSION_REF_MIN_LENGTH = 8
OPAQUE_SESSION_REF_MAX_LENGTH = 128

def is_opaque_session_ref(value: object) -> bool:
    return (
        isinstance(value, str)
        and OPAQUE_SESSION_REF_MIN_LENGTH <= len(value) <= OPAQUE_SESSION_REF_MAX_LENGTH
        and value.startswith(OPAQUE_SESSION_REF_PREFIX)
        and all(character.isascii() and (character.isalnum() or character in '_-')
                for character in value)
    )

File: crates/ferry-engine/src/contracts/session_ref.rs
// 此文件由 scripts/generate-contracts.py 生成，请勿手改。

pub const OPAQUE_SESSION_REF_PREFIX: &str = "fsr_";
pub const OPAQUE_SESSION_REF_MIN_LENGTH: usize = 8;
pub const OPAQUE_SESSION_REF_MAX_LENGTH: usize = 128;

pub fn is_opaque_session_ref(value: &str) -> bool {
    (OPAQUE_SESSION_REF_MIN_LENGTH..=OPAQUE_SESSION_REF_MAX_LENGTH).contains(&value.len())
        && value.starts_with(OPAQUE_SESSION_REF_PREFIX)
        && value
            .bytes()
            .all(|byte| byte.is_ascii_alphanumeric() || matches!(byte, b'_' | b'-'))
}
